### PR TITLE
{App Service} `az functionapp deployment source config-zip`: Increase read timeout

### LINKED ISSUE #13655: az functionapp Azure Cli deploy error (v2)
## Describe the bug

**Command Name**
`az functionapp deployment source config-zip`

**Errors:**
```
HTTPSConnectionPool(host='*************.azurewebsites.net', port=443): Read timed out. (read timeout=3)
Traceback (most recent call last):
python3.6/site-packages/urllib3/connectionpool.py, ln 387, in _make_request
    six.raise_from(e, None)
<string>, ln 2, in raise_from
python3.6/site-packages/urllib3/connectionpool.py, ln 383, in _make_request
    httplib_response = conn.getresponse()
lib/python3.6/http/client.py, ln 1331, in getresponse
    response.begin()
...
socket.timeout: The read operation timed out


python3.6/site-packages/requests/adapters.py, ln 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='*****************.scm.azurewebsites.net', port=443): Read timed out. (read timeout=3)
```

## To Reproduce:
Deploy zipped publish package to Azure Function App service. using the following command

`az functionapp deployment source config-zip -g [RESOURCE-GROUP ]-n [ FUNCTION APP NAME] --src publish.zip`

## Environment Summary
```
Linux-4.15.0-1082-azure-x86_64-with-debian-stretch-sid
Python 3.6.5
Installer: DEB

azure-cli 2.5.1 *

```

### PR DESCRIPTION
**Pull request by gukoff**
**Related command**

```sh
az functionapp deployment source config-zip --name "$(myFuncName)" --resource-group "$(myRg)" --src "myZipFile"
```

**Description**

Attempt to fix #13655.

Exact error I am seeing:

```

    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/appservice/custom.py", line 531, in enable_zip_deploy_functionapp
    remove_remote_build_app_settings(cmd, resource_group_name, name, slot)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/appservice/custom.py", line 671, in remove_remote_build_app_settings
    scm_is_up_to_date = validate_app_settings_in_scm(
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/appservice/custom.py", line 1230, in validate_app_settings_in_scm
    scm_settings = _get_app_settings_from_scm(cmd, resource_group_name, name, slot)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/appservice/utils.py", line 134, in call
    raise exception
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/appservice/utils.py", line 130, in call
    return func(*args, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/appservice/custom.py", line 1259, in _get_app_settings_from_scm
    response = requests.get(settings_url, headers=headers, auth=(username, password), timeout=3)
  File "/opt/az/lib/python3.10/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/requests/sessions.py", line 542, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/az/lib/python3.10/site-packages/requests/sessions.py", line 655, in send
    r = adapter.send(request, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/requests/adapters.py", line 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='my-function-name.scm.azurewebsites.net', port=443): Read timed out. (read timeout=3)

```

**Testing Guide**

N/A

**History Notes**

{functionapp} longer read timeout when fetching Source Control Manager (SCM) settings

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@StrawnSC Could you please help review this PR?

**Comment by StrawnSC** [comment]
Hey @kamperiadis and @kaibocai it looks like this is a change for functionapp deployments. Could you guys review this?

File: src/azure-cli/azure/cli/command_modules/appservice/utils.py [hunk 1]
```diff
@@ -128,10 +128,10 @@ def call(*args, **kwargs):
             while True:
                 try:
                     return func(*args, **kwargs)
-                except excpt_type as exception:  # pylint: disable=broad-except
+                except excpt_type:  # pylint: disable=broad-except
```
Comment (kaibocai) [inline src/azure-cli/azure/cli/command_modules/appservice/utils.py:131]: Why we update this part, besides set a longer timeout?

**Review by kaibocai — COMMENTED** [review]
Thanks for the PR, left one question. Thanks.

File: src/azure-cli/azure/cli/command_modules/appservice/utils.py (hunk 1 above)
Comment (gukoff) [reply src/azure-cli/azure/cli/command_modules/appservice/utils.py:131]: Just making the code more idiomatic. If the goal is to re-raise the exception, there's no point making a variable with it.

Also, if executing on Python2, the current code will lose the original stacktrace when re-raising the exception. 
But I'm not sure if az-cli supports running on Python 2 at all?

**Pull request merged**